src/simple_openid_connect/integrations/djangorestframework/permissions.py

```diff
@@ -41,6 +42,31 @@ def _validate_scopes(required_scopes: str, granted_scopes: str) -> bool:
             for i_scope in required_scopes.split(" ")
         )
 
+
+class HasSessionScope(_HasScope):  # type: ignore
+    def has_permission(self, request: HttpRequest, view: Any) -> bool:
+        # validate that enough information is present to authorize the request
+        if not request.user.is_authenticated:
+            logger.error(
+                "session permission is supposed to be checked but the request was not authenticated; denying access"
+            )
+            return False
+        if not hasattr(request.user, "openid"):
+            logger.error(
+                "session permission is supposed to be checked but the request was not authenticated with an OpenidSession; denying access"
+            )
+            return False
+        session_scopes = OpenidSession.objects.filter(
+            sid=request.user.openid.id_token.sid
+        ).values_list("scope", flat=True)
```

Comment:
Is there a reason you're using this lookup instead of the reverse relation from `OpenidUser` to `OpenidSession`?